### ISSUE TITLE
Truncated blob name

### ISSUE THREAD

**Issue by danielgary** [LanceMcCarthy/Action-AzureBlobUpload#122]
Hey Lance!

Was using this to upload an executable to Azure with the following name: ```./build/bin/data-composer-0.9.7.exe```.  It worked, but the blob name was ```ta-composer-0.9.7.exe```.

Any thoughts on what might have happened?

Here's the configuration I used:
```yaml
      - name: Upload to Azure Blob Storage
        uses: LanceMcCarthy/Action-AzureBlobUpload@v1.8.3
        with:
          connection_string: ${{ secrets.AZURE_BLOB_CONNECTION_STRING }}
          container_name: releases
          source_folder: ./build/bin/
```

**Comment by LanceMcCarthy** [comment]
This might have been due to a recent change in 1.8.3, if you drop down to 1.8.2 does it work again?

There seems to be a lot of inconsistencies with the path trimming, I'll spend more time on that this weekend a d come up with a more stable approach for 1.9

**Comment by danielgary** [comment]
I'll give 1.8.2 a try.  Thanks!

**Comment by LanceMcCarthy** [comment]
@danielgary 1.8.4 is now published in the GitHub Marketplace. Thanks for the help!

> Community Note - Due to the wide range of environments and path configurations, there's a slight change this update broke someone's build. Please open a new Issue and share the workflow and expected results so I can add a condition for it in the join/normalization logic.

**Comment by LanceMcCarthy** [comment]
I'm reopening this because the root cause was that you set a single file as the `source_folder` value. 

This action doesn't have support for single file uploads, it is intended to upload a **folder**'s contents (with support for recursion of subfolder contents).

I am going to add single file support in the next release, but in the meantime, I need to undo primary the changes in #123

**Comment by LanceMcCarthy** [comment]
@danielgary  I've [published v1.9.0](https://github.com/LanceMcCarthy/Action-AzureBlobUpload/releases/tag/v1.9.0) which includes a fix for this issue. Please try the new release in your workflow and let me know if you have any issues.

I did a lot of housekeeping and refactoring. Things like better tests, separation of concerns refactoring and more. These are places where I'd like to condense the verbosity, but right now it has a nice balance of readability and conciseness.. so I wont make that big shift until 2.0+.

Ultimately, this was a tricky one because a fix for one problem ended up breaking another feature. This is exactly what happened with your PR I merged, it ended up flattening out other user's destination folder structures (even all the excellent tests you wrote doesn't catch it, I needed to manually look in my test blob container to verify  :)
